### PR TITLE
Add `on_error` settings for `mysql2`

### PR DESCRIPTION
**Pull request by TonyCTHsu**
**What does this PR do?**

Add `on_error` settings for `mysql2` instrumentation.  

**Motivation:**

This proc provides the flexibility to handle exception that does not want to be reported.

**For Datadog employees:**
- [ ] If this PR touches code that signs or publishes builds or packages, or handles
credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.
- [x] This PR doesn't touch any of that.

### CODE REVIEW

**Review by ekump — APPROVED** [review]

File: docs/GettingStarted.md [hunk 1]
```diff
@@ -1219,7 +1219,7 @@ client.query("SELECT * FROM users WHERE group='x'")
...
 | `peer_service`        | `DD_TRACE_MYSQL2_PEER_SERVICE` | Name of external service the application connects to                                                                                                                                                                                                                                                                                                                    | `nil`        |
 | `comment_propagation` | `DD_DBM_PROPAGATION_MODE`      | SQL comment propagation mode  for database monitoring. <br />(example: `disabled` \| `service`\| `full`). <br /><br />**Important**: *Note that enabling sql comment propagation results in potentially confidential data (service names) being stored in the databases which can then be accessed by other 3rd parties that have been granted access to the database.* | `'disabled'` |
-
+| `on_error` | | Custom error handler invoked when raises an error. Provided `span` and `error` as arguments. Sets error on the span by default. Useful for ignoring errors that are handled at the application level. | `proc { \|span, error\| span.set_error(error) unless span.nil? }` |
```
Comment (ekump) [inline docs/GettingStarted.md:1222]:
```suggestion
| `on_error` | | Custom error handler invoked when mysql raises an error. Provided `span` and `error` as arguments. Sets error on the span by default. Useful for ignoring errors that are handled at the application level. | `proc { \|span, error\| span.set_error(error) unless span.nil? }` |
```

**Review by marcotc — APPROVED** [review]

File: docs/GettingStarted.md [hunk 2]
```diff
@@ -1219,7 +1219,7 @@ client.query("SELECT * FROM users WHERE group='x'")
...
 | `peer_service`        | `DD_TRACE_MYSQL2_PEER_SERVICE` | Name of external service the application connects to                                                                                                                                                                                                                                                                                                                    | `nil`        |
 | `comment_propagation` | `DD_DBM_PROPAGATION_MODE`      | SQL comment propagation mode  for database monitoring. <br />(example: `disabled` \| `service`\| `full`). <br /><br />**Important**: *Note that enabling sql comment propagation results in potentially confidential data (service names) being stored in the databases which can then be accessed by other 3rd parties that have been granted access to the database.* | `'disabled'` |
```
Comment (alai97) [inline docs/GettingStarted.md:1221]:
```suggestion
| `comment_propagation` | `DD_DBM_PROPAGATION_MODE`      | SQL comment propagation mode for database monitoring. <br />(example: `disabled` \| `service`\| `full`). <br /><br />**Important**: *Note that enabling SQL comment propagation results in potentially confidential data (service names) being stored in the databases which can then be accessed by other third parties that have been granted access to the database.* | `'disabled'` |
```

**Review by alai97 — APPROVED** [review]
Looks good, two suggestions!

**Comment by alekgosk** [comment]
Wow, this is amazing @TonyCTHsu, exactly what we are looking for!

Question: do you think this is something we can also implement for the grape integration?

We have this issue currently where we catch all exceptions using `rescue :all`, yet they still appear in DataDog's Error Tracking which creates unnecessary noise, we described it here = https://github.com/DataDog/dd-trace-rb/issues/3253

**Comment by TonyCTHsu** [comment]
👋 @alekgosk , We are preparing our 2.0 release, which would include this PR https://github.com/DataDog/dd-trace-rb/pull/3370, which might help

**Pull request merged**